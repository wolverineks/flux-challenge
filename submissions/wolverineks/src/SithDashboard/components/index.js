export {default as CurrentPlanetIndicator} from './CurrentPlanetIndicator.jsx'
export {default as Siths} from './Siths.jsx'
export {default as SithList} from './SithList.jsx'
export {default as Sith} from './Sith.jsx'
export {default as Button} from './Button.jsx'
export {default as ScrollButtons} from './ScrollButtons.jsx'
