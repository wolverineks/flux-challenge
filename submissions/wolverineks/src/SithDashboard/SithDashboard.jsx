// SithDashboard/SithDashboard.jsx
// @flow

import React, {Component} from 'react'

import {Button, CurrentPlanetIndicator, ScrollableList, List, Item, ScrollButtons, Sith} from './components'
import {classNames} from './styles'

import {type Planet, type Siths} from './Types'
type Props = {planet: Planet, siths: Siths}
type State = {}

export default class SithDashboard extends Component<Props, State> {
  render () {
    const {planet, siths} = this.props

    return <div className={classNames.main}>
      <div className={classNames.mainInner}>

        <CurrentPlanetIndicator planet={planet} />

        <ScrollableList>
          <List>
            <Item><Sith name={siths[0] ? siths[0].name : undefined} homeworld={siths[0] ? siths[0].homeworld : undefined} /></Item>
            <Item><Sith name={siths[1] ? siths[1].name : undefined} homeworld={siths[1] ? siths[1].homeworld : undefined} /></Item>
            <Item><Sith name={siths[2] ? siths[2].name : undefined} homeworld={siths[2] ? siths[2].homeworld : undefined} /></Item>
            <Item><Sith name={siths[3] ? siths[3].name : undefined} homeworld={siths[3] ? siths[3].homeworld : undefined} /></Item>
            <Item><Sith name={siths[4] ? siths[4].name : undefined} homeworld={siths[4] ? siths[4].homeworld : undefined} /></Item>
          </List>

          <ScrollButtons>
            <Button direction={'up'} />
            <Button direction={'down'} />
          </ScrollButtons>
        </ScrollableList>

      </div>
    </div>
  }
}
