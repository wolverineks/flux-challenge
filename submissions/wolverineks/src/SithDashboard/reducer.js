// SithDashboard/reducer.js
// @flow

import {combineReducers} from 'redux'

import {reducer as currentPlanetIndicator} from './components/CurrentPlanetIndicator'
import {reducer as scrollableList} from './components/ScrollableList'

export default combineReducers({currentPlanetIndicator, scrollableList})
