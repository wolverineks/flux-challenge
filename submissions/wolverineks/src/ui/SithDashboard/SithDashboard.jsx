// @flow

import React from 'react'

import {Header, Body, Button, Planet, ScrollableList, List, Item, ScrollButtons, Sith} from './components'
import {name, homeworld} from './utils'
import {classNames} from '.'
import {type Siths, type Planet as PlanetType} from '../../state/types'
import {List_} from '../../state/connectors'
type Props = {siths: Siths, planet: PlanetType}

const SithDashboard = ({siths, planet}: Props) =>
  <div className={classNames.main}>
    <div className={classNames.mainInner}>

      <Header>
        <Planet planet={planet} />
      </Header>

      <Body>
        <List_ render={(list) => {
          <ScrollableList data={siths} render={() => <div>
            <List list={list}>
              <Item><Sith name={name(list[0])} homeworld={homeworld(list[0])} /></Item>
              <Item><Sith name={name(list[1])} homeworld={homeworld(list[1])} /></Item>
              <Item><Sith name={name(list[2])} homeworld={homeworld(list[2])} /></Item>
              <Item><Sith name={name(list[3])} homeworld={homeworld(list[3])} /></Item>
              <Item><Sith name={name(list[4])} homeworld={homeworld(list[4])} /></Item>
            </List>

            <ScrollButtons render={() => <div>
              <Button direction={'UP'} />
              <Button direction={'DOWN'} />
            </div>} />
          </div>} />
        }} />
      </Body>

    </div>
  </div>

export default SithDashboard
