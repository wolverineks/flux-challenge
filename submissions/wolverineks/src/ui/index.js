// @flow

import type {Children} from '../state/types'
import {Layout, Header, Body, PlanetIndicator, Sith, ScrollableList, ScrollButtons, ScrollButton, List, Item} from './components'
import classNames from './styles'

export type {Children}
export {
  Layout,
  Header,
  Body,
  PlanetIndicator,
  Sith,
  ScrollableList,
  ScrollButtons,
  ScrollButton,
  List,
  Item,
  classNames
}
export default Layout
