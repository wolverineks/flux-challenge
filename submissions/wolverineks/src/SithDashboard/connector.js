// SithDashboard/SithDashboardConnector
// @flow

import {connect} from 'react-redux'

import SithDashboard from './SithDashboard.jsx'
import * as SELECTORS from './selectors'

import type {Dispatch} from '../types'
import type {State, Planet, Siths} from './types'

export type StateProps = {planet: Planet, siths: Siths}
export type DispatchProps = {dispatch: Dispatch}
export type Props = StateProps & DispatchProps

const mapStateToProps = (state: State): StateProps => ({
  planet: SELECTORS.getPlanet(state),
  siths: SELECTORS.getSiths(state)
})
const mapDispatchToProps = (dispatch: Dispatch): DispatchProps => ({
  dispatch
})

export default connect(mapStateToProps, mapDispatchToProps)(SithDashboard)
