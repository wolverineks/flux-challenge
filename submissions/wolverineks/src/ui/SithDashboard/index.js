// @flow

import SithDashboard from './SithDashboard.jsx'
import {Header, Body, Planet, Sith, ScrollableList, ScrollButtons, Button, List, Item} from './components'
import classNames from './styles'

export {
  SithDashboard,
  Header,
  Body,
  Planet,
  Sith,
  ScrollableList,
  ScrollButtons,
  Button,
  List,
  Item,
  classNames
}
export default SithDashboard
