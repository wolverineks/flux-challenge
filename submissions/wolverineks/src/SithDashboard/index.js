// SithDashboard/index.js
// @flow

import selectors from './selectors'
import reducer from './reducer'

import Main from './SithDashboard.jsx'
import connector from './connector'

export {
  Main,
  reducer,
  selectors,
}

export default connector
