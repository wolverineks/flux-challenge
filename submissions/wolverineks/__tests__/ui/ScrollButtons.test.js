// @flow

import React from 'react'
import renderer from 'react-test-renderer'
import {ScrollButtons, Button, classNames} from '../../src/ui/SithDashboard'

describe('ScrollButtons', () => {
  describe('When has buttons', () => {
    test('ScrollButtons', () => {
      const className = classNames.scrollButtons
      const component = renderer.create(<ScrollButtons className={className}>
        <Button direction={'UP'} />
        <Button direction={'DOWN'} />
      </ScrollButtons>)
      const tree = component.toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
