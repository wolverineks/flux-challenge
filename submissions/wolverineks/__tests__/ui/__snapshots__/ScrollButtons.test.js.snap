// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollButtons When has buttons ScrollButtons 1`] = `
<div
  className="css-scroll-buttons"
>
  <button
    className="css-button-up"
    onPress={undefined}
  />
  <button
    className="css-button-down"
    onPress={undefined}
  />
</div>
`;
